### PR TITLE
MM - 19155 Change plugin helpers files to be golint complied

### PR DESCRIPTION
**Pull request by andresoro**
#### Summary
This PR makes files under plugin/helpers* compliant with the golint command and also creates a new makefile target to check said compliance. 
#### Ticket Lin
  Fixes [Issue 12657](https://github.com/mattermost/mattermost-server/issues/12657)

https://mattermost.atlassian.net/browse/MM-19155

### CODE REVIEW

**Comment by andresoro** [comment]
Got build error ' Blocked due to plan-no-user-seats-available '. Unsure why.

**Comment by YannHulot** [comment]
@andresoro this is circle ci specific issue. From a thread I read, it seems to happen randomly. It's currently happening on my MR as well.

File: plugin/helpers.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ package plugin
 
 import "github.com/mattermost/mattermost-server/model"
 
+// Helpers interface that encompasses methods for plugin helpers to ensure bots and interact with data store
```
Comment (hanzei) [inline plugin/helpers.go]: I think we can generalise this a bit more. It's not just about bots and the store. Helpers in general abstract common patterns away from plugin devs.

Would you also please add:

`Plugins obtain access to the Helpers by embedding MattermostPlugin.`

**Comment by ali-farooq0** [comment]
Thanks for your contribution @andresoro! 🎉 Would you have time to address @hanzei's concerns above?

**Comment by hanzei** [comment]
Will request QA review when the Dev review is done

**Comment by hanzei** [comment]
@andresoro I added the changes myself because I want to merge this soon. Hope you are fine with this.

@ali-farooq0 Please take a second look.

**Comment by hanzei** [comment]
@prapti This is ready for review

**Pull request merged**